src/blockdev.rs

```diff
@@ -123,14 +157,28 @@ fn get_partitions(device: &str) -> Result<Vec<BlkDev>> {
         // parse key-value pairs
         let fields = split_lsblk_line(line);
         if let Some(name) = fields.get("NAME") {
-            // Skip the device itself
-            if device == name {
+            // Only return partitions.  Skip the whole-disk device, as well
+            // as holders like LVM or RAID devices using one of the partitions.
+            if fields.get("TYPE") != Some(&"part".to_string()) {
                 continue;
             }
+            let (mountpoint, swap) = match fields.get("MOUNTPOINT") {
+                Some(mp) => {
```

Comment:
Not sure if it reads better, but I think here you could employ a [match guard](https://doc.rust-lang.org/stable/rust-by-example/flow_control/match/guard.html):
`Some(mp) if mp == "XYZ" => { ... }`